2016-09/september-07.md

```diff
@@ -0,0 +1,80 @@
+	* Sean describes this "hackyness" is when a loader accesses `compiler` properties from `this` such as `this._compiler` to execute plugin based functionality. Needs to be deprecated.
+	* Tobias mentions that maybe we can add functionality to the loader API to allow this to work. Sean explained that the reason these TS loaders need plugin access is because it needs to defer typechecking until the `compilation` is finished. 
+	* Tobias mentions that deferred execution could be a feature we need to add to the loader API. Could also be useful for image sprites, etc. 
+	* Johannes brings up that this is the same problem to `extract-text-plugin` (when you need to apply a plugin and a loader together).
+	* Tobias states that this could fix a great deal of "hackyness" from `extract-text-plugin` and move it to a loader. 
+	* Sean suggested that the deferred execution be expressed by unwrapping a promise. (
+	The loader function itself would return the promise. ) Would be minimal additional effort and easy for developers to understand.
+	* Tobias mentioned that tool that would be moved and use the new deferred execution API [(doesn't exist yet but suggested idea) `image-sprite-loader` and the `seperate-css-loader`] would need to include an `invalidate` method to issue recompilation from the callback. It would also need to include the following: 	
+		* errors/warnings
+		* invalidating => recompilation w/ new data
+		* emitting files
+		* access to the current chunk info (edited)
+	* additionally noted it would fall after optimization and before id creation (in the compilation step)
+	* This would also allow ts-loader and awesome-typescript-loader to execute typechecking in the resolve promise. (allowing tools like HappyPack to execute loaders parallel/multithreaded)
+	* Johannes mentions that it would be important that we get feedback from loader authors to see if this would help solve some of their issues (including HappyPack).
+	* Tobias described, (once implementing this feature [based upon the comments in the linked issue]) that a `disk-cache-loader` would look similar to having the following.
+		* pithing phase (this is a pitching loader)
+		* check disk cache for an entry
+		* if there is an entry:
+		* hash all content of entry.dependencies
+		* compare hash with entry
+		* if equal:
+		   * return entry.source and entry.dependencies as result
+		* continue loader execution
+		normal phase:
+		* store dependencies, hash of dependencies and source in disk cache (edited)
+		* alternativly it could use loader-runner to run the remaining loaders. This would allow to cache more data: emitted files, errors, warnings. This would not be possible in the normal loader chain.
+	* Issues will be created for each of these missing features. (Sean).
```

Comment:
There's one issue/defect missing here: We are planing to deprecate all properties and methods that make a multi-thread compilation impossible, that is:

- No synchronous APIs (`resolveSync`)
- No shared state (`_compiler`)

Could you add that @TheLarkInn ?